access-common/src/main/java/edu/unc/lib/dl/ui/service/SolrQueryLayerService.java

```diff
@@ -1014,26 +1017,24 @@ public void setCollectionsPid(PID collectionsPid) {
 	 * @return the count, or -1 if there was an error retrieving the count
 	 */
 
-	public int getDepartmentsCount() {
-
-		SolrQuery query;
-		QueryResponse response;
-
-		query = new SolrQuery();
-		query.setQuery("*:*");
-		query.setRows(0);
-		query.addFacetField("department");
-		query.setFacetLimit(-1);
-
+	public int getDepartmentsCount(AccessGroupSet accessGroups) {
 		try {
-			response = this.executeQuery(query);
+			StringBuilder queryBuilder = new StringBuilder("*:*");
+			addAccessRestrictions(queryBuilder, accessGroups);
+
+			SolrQuery query = new SolrQuery();
+			query.setQuery(queryBuilder.toString());
+			query.setRows(0);
+			query.addFacetField("department");
+			query.setFacetLimit(-1);
+			
+			QueryResponse response = this.executeQuery(query);
 			return response.getFacetField("department").getValueCount();
-		} catch (SolrServerException e) {
-			LOG.error("Error retrieving Solr object request: " + e);
+		} catch (SolrServerException | AccessRestrictionException e) {
+			LOG.error("Error retrieving department list", e);
 		}
 
 		return -1;
```

Comment:
can this return statement be moved inside the catch-block, or will the compiler complain? i think it would make for clearer code to move it inside, since that return value only gets returned if the catch-block is executed